multicluster/controllers/multicluster/commonarea/remote_common_area_manager.go

```diff
@@ -112,50 +104,49 @@ func NewRemoteCommonAreaManager(clusterSetID common.ClusterSetID, clusterID comm
  */
 
 func (r *remoteCommonAreaManager) Start() error {
-	klog.InfoS("Starting RemoteCommonAreaManager", "ClusterSet", r.clusterSetID)
+	klog.InfoS("Starting RemoteCommonAreaManager", "clusterset", r.clusterSetID)
 	stopCtx, stopFunc := context.WithCancel(context.Background())
 	r.stopFunc = stopFunc
 
-	// Start a Timer for every 5 seconds when leader election is
-	// performed, if necessary.
+	// Start a Timer for every 5 seconds to check leader connection and
+	// update active leader if necessary.
 	ticker := time.NewTicker(5 * time.Second)
 
 	go func() {
 		for {
 			select {
 			case <-stopCtx.Done():
-				klog.InfoS("Stopping RemoteCommonAreaManager", "ClusterSet", r.clusterSetID)
+				klog.InfoS("Stopping RemoteCommonAreaManager", "clusterset", r.clusterSetID)
 				close(r.eventChan)
 				for _, rc := range r.remoteCommonAreas {
 					rc.Stop()
 				}
 				r.remoteCommonAreas = nil
-				// we are done.
 				ticker.Stop()
 				return
 			case event := <-r.eventChan:
-				// process the event.
 				clusterID := event.remoteCommonArea.GetClusterID()
-
 				if event.isAdd {
 					if _, ok := r.remoteCommonAreas[clusterID]; ok {
-						klog.Warningf("Cannot add RemoteCommonArea that is already present", "Cluster", clusterID)
+						klog.Warningf("Cannot add RemoteCommonArea that is already present", "cluster", clusterID)
 					} else {
-						klog.InfoS("Adding and starting RemoteCommonArea", "Cluster", clusterID)
+						klog.InfoS("Adding and starting RemoteCommonArea", "cluster", clusterID)
 						event.remoteCommonArea.Start()
 						r.remoteCommonAreas[clusterID] = event.remoteCommonArea
 					}
 				} else {
 					if rc, ok := r.remoteCommonAreas[clusterID]; !ok {
-						klog.Warningf("Cannot remove RemoteCommonArea that is not present", "Cluster", clusterID)
+						klog.Warningf("Cannot remove RemoteCommonArea that is not present", "cluster", clusterID)
 					} else {
-						klog.InfoS("Removing and stopping RemoteCommonArea", "Cluster", clusterID)
+						klog.InfoS("Removing and stopping RemoteCommonArea", "cluster", clusterID)
 						rc.Stop()
 					}
 					delete(r.remoteCommonAreas, clusterID)
 				}
 			case <-ticker.C:
-				r.RunLeaderElection()
+				// Check the leader connection status periodically,
```

Comment:
You did not reply to my question - do we have a way to be notified synchronously or not at connection status changes, instead of doing periodical check (I am not asking the overhead of the check)?

Reply:
The connection status change detection in current implementation is actually [a go routine](https://github.com/antrea-io/antrea/blob/main/multicluster/controllers/multicluster/commonarea/remote_common_area.go#L321-L347) to update status periodically. It will try to update the member announce to the leader every 5 seconds, and change `connected` to false when there is any error happened. We may add a notification chan between remote common area manager and remote common area to be notified, but I feel it's a little overload considering the status change is also happened periodically.